### PR TITLE
Fix errors encountered when auditing various profiles

### PR DESCRIPTION
**Pull request by victorporof**
Signed-off-by: Victor Porof <victorporof@google.com>

**Summary**
Fixes a few errors encountered while using the `analyzeTrace` API. The whole `NetworkRecordsMaker` implementation works decently but (as it is known), it requires polish, which is out of the scope of this PR. I just wanted to keep things running.

cc @paulirish

### CODE REVIEW

File: lighthouse-core/lib/network-records-from-trace.js [hunk 1]
```diff
@@ -60,16 +60,16 @@ class NetworkRecordsMaker {
...
+        console.log('missing some trace events.... skipping :(');
         continue;
       }
 
-      // If we have an incomplete set of events here, we choose to drop the network
-      // request rather than attempt to synthesize the missing data.
-      if (!evtBag.sendRequest || !evtBag.receiveResponse) {
-        console.log('missing some trace events.... skipping :(')
+      if (evtBag.sendRequest.args.data.url?.startsWith('data:')) {
+        // TODO: handle the fact that data uris dont ahve a timing block at all :/
```
Comment (patrickhulce) [inline lighthouse-core/lib/network-records-from-trace.js:71]:
```suggestion
        // TODO: handle the fact that data uris don't have a timing block at all :/
```

**Review by patrickhulce — COMMENTED** [review]
feels weird for me to review a PR to a branch I didn't know existed, containing JS features we don't support yet 😅 

but LGTM, @paulirish this appears to be your experimental branch?

**Comment by paulirish** [comment]
> but LGTM, @paulirish this appears to be your experimental branch?

bingo. it's the "don't need to use devtools log for network requests, instead we use the trace" branch


> feels weird for me to review a PR to a branch I didn't know existed, containing JS features we don't support yet 😅


heh. i'm gonna merge this PR for now but will address the `?.` and `??` before I PR it against master.

**Comment by victorporof** [comment]
Thanks!

**Pull request merged**